proc/process.c

```diff
@@ -1328,3 +1446,31 @@ int _process_init(vm_map_t *kmap, vm_object_t *kernel)
 	hal_exceptionsSetHandler(EXC_UNDEFINED, process_illegal);
 	return EOK;
 }
+
+
+void tls_init(tls_t *dest, tls_t *source, vm_map_t *map)
+{
+	dest->tdata_sz = source->tdata_sz;
+	dest->tbss_sz = source->tbss_sz;
+	dest->tls_sz = round_page(source->tls_sz);
+	dest->arm_m_tls = source->arm_m_tls;
+
+	dest->tls_base = vm_mmap(map, NULL, NULL, dest->tls_sz, PROT_READ | PROT_WRITE | PROT_USER, NULL, 0, MAP_NONE);
+
+	if (dest->tls_base != NULL) {
```

Comment:
We should return -ENOMEM if it was indeed NULL - we shouldn't "successfully" spawn a thread that expects TLS without TLS